### PR TITLE
[DOC] Add note for CM creation in before/after load summaries (CC)

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- Documentation

### Description

As we discussed offline, it could be good to have some note in documentation about the creation of CM when we are deploying the `KafkaRebalance`, because ATM it seems that (from my POV) CM with summaries is created together with the KR. I added a note about this so it will be more clearer for all.

### Checklist

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

**Review by PaulRMellor — APPROVED** [review]

File: documentation/modules/cruise-control/con-optimization-proposals.adoc [hunk 1]
```diff
@@ -113,6 +113,8 @@ The broker load is stored in a ConfigMap (with the same name as the KafakRebalan
 Each metric consist of three values.
 The first is the metric value before the optimization proposal is applied, the second is the expected value of the metric after the proposal is applied, and the third is the difference between the first two values (after minus before).
 
+NOTE: The ConfigMap will appear once the KafkaRebalance is in `ProposalReady` state and remain even after the rebalancing process.
```
Comment (laidan6000) [inline documentation/modules/cruise-control/con-optimization-proposals.adoc:116]:
```suggestion
NOTE: The ConfigMap appears when the KafkaRebalance resource is in the `ProposalReady` state and remains after the rebalance is complete.
```

**Comment by laidan6000** [comment]
Please fix the typo on line 112: KafakRebalance custom resource. Thanks!

**Comment by laidan6000** [comment]
Please ensure this change is cherry-picked into the `release-0.24.x` branch.

File: documentation/modules/cruise-control/con-optimization-proposals.adoc (hunk 1 above)
Comment (im-konge) [reply documentation/modules/cruise-control/con-optimization-proposals.adoc]: thanks! :)

**Comment by im-konge** [comment]
@laidan6000 we discussed this with @scholzj - typo fix is in #5181 (will be cherry-picked into the `release-0.24.x`) and this note will go to the `0.25.0` - because last fix from @ShubhamRwt is present on current `main` branch and not cherry-picked to the `release-0.24.x`

**Comment by scholzj** [comment]
> Please ensure this change is cherry-picked into the release-0.24.x branch.

This doesn't really apply to 0.24. In 0.24 the CM is deleted when the Proposal is approved or somethinf like that.

**Comment by laidan6000** [comment]
@im-konge - That makes sense, thanks. Sorry for any confusion!

**Comment by im-konge** [comment]
@laidan6000 no problem! :)

**Review by tomncooper — APPROVED** [review]
LGTM

**Pull request merged**